### PR TITLE
Check for host and port before checking network

### PR DESCRIPTION
**Pull request by chStaiger**
Small bugfix when irods_environment does not contain minimal information.

### CODE REVIEW

File: ibridges/session.py [hunk 1]
```diff
@@ -138,8 +138,10 @@ def _network_check(self, hostname: str, port: int) -> bool:
     def connect(self) -> iRODSSession:
         """Establish an iRODS session."""
         irods_host = self._irods_env.get('irods_host', '')
-        irods_port = int(self._irods_env.get('irods_port', ''))
-        network = self._network_check(irods_host, irods_port)
+        irods_port = self._irods_env.get('irods_port', '')
```
Comment (qubixes) [inline ibridges/session.py:141]: I would use `None` as the default value for when it's missing

File: ibridges/session.py [hunk 2]
```diff
@@ -138,8 +138,10 @@ def _network_check(self, hostname: str, port: int) -> bool:
...
+        if irods_host == '' or irods_port == '':
+            raise LoginError("No host or port set in environment file.")
```
Comment (qubixes) [inline ibridges/session.py:143]: I would put this in the network check itself.

File: ibridges/session.py [hunk 3]
```diff
@@ -293,6 +295,8 @@ def _translate_irods_error(exc) -> Exception:  # pylint: disable=too-many-return
             return LoginError("Host, port, irods_client_server_policy or "
                               "irods_client_server_negotiation not set correctly in "
                               "irods_environment.json")
+    if isinstance(exc, TypeError):
```
Comment (qubixes) [inline ibridges/session.py:298]: This is kind of horrible, but I suppose there's not much we can do on our side here?

**Review by qubixes — CHANGES_REQUESTED** [review]
A nice feature for users to debug why things are not going as planned!

File: ibridges/session.py (hunk 3 above)
Comment (chStaiger) [reply ibridges/session.py:298]: The TypeError changes its message depending on which key in the iRODS file is missing. So I could not think of a more elegant way to get this message passed on :(

**Pull request merged**